### ISSUE TITLE
[Bug] Using `change:reload` with `--disabled` creates a delayed preview trigger

### ISSUE THREAD

**Issue by ibhagwan** [junegunn/fzf#3311]
- [X] I have read through the manual page (`man fzf`)
- [X] I have the latest version of fzf
- [X] I have searched through the existing issues

## Info

- OS
    - [X] Linux
    - [ ] Mac OS X
    - [ ] Windows
    - [ ] Etc.
- Shell
    - [ ] bash
    - [X] zsh
    - [ ] fish

## Problem / Steps to reproduce

Consider the following command:
```sh
echo "" | fzf --ansi \
  --bind="change:reload(rg --color=always --column --line-number --smart-case {q})" \
  --preview="echo {}" --preview-window="nohidden:down,50%" \
  --disabled
```
Now try to search for anything, with the 1st keypress the preview command does not trigger, with the 2nd keypress the preview command **triggers with the previous result set selection**.

> Removing `--disabled` solves the issue but messes up with the highlighting/ordering (obviously).

![Peek 2023-05-26 10-20](https://github.com/junegunn/fzf/assets/59988195/842565a4-3884-43d0-9f3a-862b6fcd8f0d)

**Comment by ibhagwan** [comment]
Just did a bit more testing with older versions, this bug was first introduced with `0.40.0`, version `0.39.0` works as expected.

**Comment by junegunn** [comment]
Released 0.41.1 with the fix.

**Comment by nyngwang** [comment]
Just built it, thank you for the swift fix!

<img src="https://github.com/junegunn/fzf/assets/24765272/e5e68c63-bc09-493f-8bba-ccd9b7239ef1" width="400">

**Comment by ibhagwan** [comment]
> Released 0.41.1 with the fix.

Ty @junegunn for the quick turnaround.
